### PR TITLE
chore(samples): adjust examples to support connecting to cloud

### PR DESCRIPTION
**Pull request by pihme**
## Description

Adjusts examples so that they can be used with cluster in Camunda Cloud.

Change set:
- add JavaDocs explaining which environment variables need to be set to connect to Camunda Cloud
- add logic to support both local deployments and cloud deployments

## Related issues

https://github.com/camunda-cloud/camunda-cloud-documentation/issues/88

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [X] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
 # Zeebe Java Client Examples
 
-This Maven project contains a number of examples each performing a specific task using the Zeebe Java client. For an overview and context have a look at the Zeebe documentation: https://docs.zeebe.io/java-client-examples/README.html
\ No newline at end of file
+This Maven project contains a number of examples each performing a specific task using the Zeebe
+Java client.
+
+For an overview and context have a look at
+the [Zeebe documentation](https://docs.camunda.io/docs/product-manuals/clients/java-client-examples/index)
```
Comment (korthout) [inline samples/README.md:7]: This seems to be a dead link. I wonder whether index should be in the url

File: samples/src/main/java/io/zeebe/example/cluster/TopologyViewer.java [hunk 2]
```diff
@@ -11,16 +11,47 @@
 import io.zeebe.client.ZeebeClientBuilder;
 import io.zeebe.client.api.response.Topology;
 
+/**
+ * Sample application that connects to a cluster on Camunda Cloud, or a locally deployed cluster.
```
Comment (korthout) [inline samples/src/main/java/io/zeebe/example/cluster/TopologyViewer.java:15]: Same suggestion as in https://github.com/zeebe-io/zeebe-get-started-java-client/pull/6

**Review by korthout — COMMENTED** [review]
Generally looks good, just a few comments

File: samples/README.md (hunk 1 above)
Comment (pihme) [reply samples/README.md:7]: It is rather a not yet alive link. It will work when we update the documentation.

And yes, index needs to be in the URL

**Comment by pihme** [comment]
Merge should wait until new documentation is released. Otherwise, some links are not yet available.

**Review by korthout — APPROVED** [review]

**Comment by pihme** [comment]
@npepinpe Any idea how to tell SonarCloud to ignore the example code in duplicate code checks?

**Comment by npepinpe** [comment]
Afaik you can configure sonarcloud via maven, so I suppose it could be done the samples pom, though I don't know how flexible it is. We could also disable it for samples if it causes too much issues.

**Pull request merged**